### ISSUE TITLE
External audio autoload

### ISSUE THREAD

**Issue by oneoneleven** [mpv-player/mpv#967]
*(haven't found anything related in issue tracker)*

It would be nice to have external audio tracks loaded automatically, like subs.

**Comment by swamp-fish** [comment]
Its a very useful option and is present in MPC-HC. Most of the TV series/Anime/Movies aren't local made and while subtitles are great sometimes you just want to relax from reading especially if you're watching it with someone else.
Currently autloading is working for subtitles.

Linux users unlike Windows ones can't switch to MPC-HC for TV series watch.

There can be 2 options:
1) Automaticly load external audio file in the same directory
2) -- in the next 2 directories(at max) as well(video.mp4 sound_folder
a few real cases:
sound_folder contains sound1_folder sound2_folder, each contains sound files
sound_folder contains sound files
)

Also 2nd option can be added to subtitles as well. At this point we have a need either to copy them or make links to the same folder as video.

Its a very needed feature please make it be.

**Comment by ebvjr** [comment]
Please mux it if that's the case using mkv/webm or any media container.

**Comment by swamp-fish** [comment]
Sure, muxing it is much faster and won't require additional disk space, much easier than tapping a play button or even manually loading audio track.

**Comment by eternal-sorrow** [comment]
Sure, if I'd make a release, I'd mux it into mkv, but if I just downloaded from tracker new season of my favourite anime with external audio, i just want to press Enter and enjoy. But I got it working for me, so it's nothing to complain about :)

**Comment by oneoneleven** [comment]
We're all glad it's working *for you*.

**Comment by swamp-fish** [comment]
For those who are interested in this feature, bomi have it now
https://github.com/xylosper/bomi/issues/189

**Comment by eternal-sorrow** [comment]
@oneoneleven, my patch was not accepted, as you can see.

**Comment by wm4** [comment]
>@oneoneleven, my patch was not accepted, as you can see.

You don't need to be so sour about it. Anyone would reject it, because it duplicated large amounts of code.

**Comment by eternal-sorrow** [comment]
It's just proof-of-concept patch. I'm not Linus Torvalds nor Dennis Ritchie. I made it in 20 minutes with the cup of coffee. And it works! Even though some major releases of mpv passed, it still works. If you can understand what this patch does, you can do the same things right way.

**Comment by oneoneleven** [comment]
Great!

Also it would be great to honor `alang` option after loading external audio files.

**Comment by wm4** [comment]
It should be doing that, though the external tracks always preferred over internal.

**Comment by hrzhu** [comment]
@wm4 Is it possible to set mpv to prefer internal audio tracks. I need this feature because I want mpv to auto-load audio commentary but I don't want mpv to use audio commentary by default.

**Comment by wm4** [comment]
Just use ``aid=1`` in your config. This forces always the first audio track (so language selection etc. is disabled), but on the other hand you probably don't need it if you use external audio tracks often.

**Comment by hrzhu** [comment]
I don't use external audio tracks often. It's just some of my anime collections have external audio commentaries. I want them to be auto-loaded so that I can toggle audio when I need them. `aid` overrides `alang` which is not what I want. I might end up muxing them into the containers.

**Comment by POMATu** [comment]
Hi guys,
I had the same issue and got really annoyed cause I usually get subs from other place and that's named differently

So I wrote the tool that autofinds and loads differently named subs and external audio by the episode number
https://github.com/POMATu/anime-tool
Hope that helps

**Comment by Lothakim** [comment]
The official manual has given this option.
Just add `–audio-file-auto=exact` in the config file `mpv.conf`.
This will auto load the audio file which has the same name with the video file in the same directory.

**Comment by POMATu** [comment]
Basically about my last comment here:

I tried implementing autoload with arbitary filenames, and I run into problem with humanities random naming. Its not possible to predict how they gonna name their files to detect the correct episode order for subs/audio taken from other source. It cant be sovled with simple regexp and I failed to do that with some lazy algorithms

Sometimes they use even incorrect Roman numbers like (IIII instead of IV) or any other random naming you cant even imagine. So this is seriously not a job for mpv developers. Its a job for some separate AI project or addon that gonna learn on big data from some torrent trackers to detect which episode match which subtitles taken from other source and etc.

So I suppose MPV player should have only an option for exact file names like any other player has and thats all.
